EventFilter/CSCRawToDigi/src/CSCComparatorData.cc

```diff
@@ -190,15 +193,16 @@ std::vector<CSCComparatorDigi> CSCComparatorData::comparatorDigis(int layer) {
 }
 
 void CSCComparatorData::add(const CSCComparatorDigi& digi, int layer) {
-  //FIXME do flipping
   int strip = digi.getStrip();
-  int halfStrip = (strip - 1) * 2 + digi.getComparator();
-  int cfeb = (strip - 1) / 16;
-  int distrip = ((strip - 1) % 16) / 2;
+  int halfstrip = digi.getHalfStrip();
+  int cfeb = digi.getCFEB();
+  int distrip = digi.getDiStrip();
 
-  // assert(distrip < 8 && cfeb < 6 && halfStrip < 161);
-  ///!!! Do we need to introduce format version here to accomodate 7 CFEBs
-  assert(distrip < 8 && cfeb < 8 && halfStrip < 225);
+  // Check the distrip and half-strip number
+  assert(cfeb < CSCConstants::MAX_CFEBS);
```

Comment:
Shouldn't this be `MAX_CFEBS_RUN2` instead?

Reply:
So actually, this function is called by this snippet here: https://github.com/cms-sw/cmssw/blob/7e8681ef6d4bdb908afd0e506555593796ab0d8b/EventFilter/CSCRawToDigi/src/CSCDigiToRaw.cc#L245-L252. It is used for the Run-1 Triple-ganged case. Relaxing it to MAX_CFEBS_RUN2 won't hurt I think, but keeping it at MAX_CFEBS should be okay. The function was superseded by https://github.com/cms-sw/cmssw/blob/7e8681ef6d4bdb908afd0e506555593796ab0d8b/EventFilter/CSCRawToDigi/src/CSCDigiToRaw.cc#L235-L244 for the Run-2 case.